packages/fxa-auth-server/lib/routes/auth-schemes/mfa.ts

```diff
@@ -97,17 +98,19 @@ export const strategy = (
 
       const sessionToken = await getCredentialsFunc(decoded.stid);
       if (!sessionToken) {
-        throw AppError.invalidToken('Parent session token not found!');
+        throw AppError.unauthorized('Token not found');
       }
 
-      // Check the underlying session
+      if (sessionToken.uid !== decoded.sub) {
+        throw AppError.unauthorized('Token invalid');
+      }
+
+      // Decorate session token with scope
+      sessionToken.scope = decoded.scope;
+
       // Finalize auth
       return h.authenticated({
-        credentials: {
-          ...sessionToken,
-          uid: decoded.sub,
-          scope: decoded.scope,
-        },
+        credentials: sessionToken,
```

Comment:
It may be helpful to add a comment  to explain this preserves getters like `authenticatorAssuranceLevel()` (the motivation for this fix), as future insurance against this breaking again?